fighters/common/src/general_statuses/dash.rs

```diff
@@ -199,20 +197,21 @@ unsafe extern "C" fn status_dash_main_common(fighter: &mut L2CFighterCommon, arg
             enable_bidou = true;
         }
     }
+
     if fighter.global_table[DASH_CALLBACK].get_bool() {
         let callable: extern "C" fn(&mut L2CFighterCommon) -> L2CValue = std::mem::transmute(fighter.global_table[DASH_CALLBACK].get_ptr());
         interrupt_if!(callable(fighter).get_bool());
     }
 
-    if ControlModule::check_button_release(fighter.module_accessor, *CONTROL_PAD_BUTTON_CSTICK_ON) {
-        // prevent game from thinking you are inputting a dashback on the frame the cstick stops overriding left stick (0.625 -> -1.0)
-        VarModule::on_flag(fighter.battle_object, vars::common::DISABLE_BACKDASH);
-        WorkModule::unable_transition_term(fighter.module_accessor, *FIGHTER_STATUS_TRANSITION_TERM_ID_CONT_TURN_DASH);
-    }
-    if stick_x == 0.0 {
-        // if you return stick to neutral after a cstick dash, allow dashbacks again
-        VarModule::off_flag(fighter.battle_object, vars::common::DISABLE_BACKDASH);
-        WorkModule::enable_transition_term(fighter.module_accessor, *FIGHTER_STATUS_TRANSITION_TERM_ID_CONT_TURN_DASH);
+    if fighter.global_table[CURRENT_FRAME].get_i32() == 0 && stick_x.abs() >= dash_stick_x {
+        // apply speed on f1 of dash (takes effect on f2 ingame)
+        let prev_speed = VarModule::get_float(fighter.battle_object, vars::common::CURR_DASH_SPEED);
+        let applied_speed = (dash_speed * PostureModule::lr(fighter.module_accessor)) + (stick_x.signum() * ((run_accel_mul + (run_accel_add * stick_x.abs())))) + prev_speed;  // initial dash speed + 1f of run acceleration + previous status' last speed
+        //println!("Changing current dash speed: {}", applied_speed);
+        let applied_speed_clamped = applied_speed.clamp(-run_speed_max, run_speed_max);
+        fighter.clear_lua_stack();
+        lua_args!(fighter, FIGHTER_KINETIC_ENERGY_ID_CONTROL, applied_speed_clamped);
+        app::sv_kinetic_energy::set_speed(fighter.lua_state_agent);

```

Comment:
moved from end of main loop to beginning, so energy gets applied before transitioning into other statuses (energy funcs run before main loop anyway)